metricbeat/internal/metrics/cpu/metrics.go

```diff
@@ -0,0 +1,217 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package cpu
+
+import (
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/metricbeat/internal/metrics"
+)
+
+// CPU manages the CPU metrics from /proc/stat
+// If a given metric isn't available on a given platform,
+// The value will be null. All methods that use these fields
+// should assume that any value can be null.
+// The values are in "ticks", which translates to milliseconds of CPU time
+type CPU struct {
+	User    metrics.OptUint `struct:"user"`
+	Sys     metrics.OptUint `struct:"system"`
+	Idle    metrics.OptUint `struct:"idle"`
+	Nice    metrics.OptUint `struct:"nice"`    // Linux, Darwin, BSD
+	Irq     metrics.OptUint `struct:"irq"`     // Linux and openbsd
+	Wait    metrics.OptUint `struct:"iowait"`  // Linux and AIX
+	SoftIrq metrics.OptUint `struct:"softirq"` // Linux only
+	Stolen  metrics.OptUint `struct:"steal"`   // Linux only
```

Comment:
should we add `,omitempty` to the fields that are not present on all systems.

Reply:
This struct isn't being directly passed upstream, but we might want to add it for the sake of future-proofing.